File: app/src/main/java/com/majeur/psclient/model/DexPokemon.java
package com.majeur.psclient.model;

import java.util.List;

public class DexPokemon extends BasePokemon {

    public int num;
    public String firstType;
    public String secondType;
    public Stats baseStats;
    public List<String> abilities;
    public String hiddenAbility;
    public float height;
    public float weight;
    public String color;
    public String gender;
    public String tier;

    public DexPokemon(String rawSpecies, int num, String firstType, String secondType,
                      Stats baseStats, List<String> abilities, String hiddenAbility,
                      float height, float weight, String color, String gender, String tier) {
        super(rawSpecies);
        this.num = num;
        this.firstType = firstType;
        this.secondType = secondType;
        this.baseStats = baseStats;
        this.abilities = abilities;
        this.hiddenAbility = hiddenAbility;
        this.height = height;
        this.weight = weight;
        this.color = color;
        this.gender = gender;
        this.tier = tier;
    }
}

File: psclient/src/main/java/com/majeur/psclient/model/pokemon/DexPokemon.kt
package com.majeur.psclient.model.pokemon

import com.majeur.psclient.model.common.Stats
import com.majeur.psclient.util.toId


class DexPokemon : BasePokemon() {

    var num = 0
    lateinit var firstType: String
    var secondType: String? = null
    lateinit var baseStats: Stats
    var abilities = emptyList<String>()
    var hiddenAbility: String? = null
    var height = 0f
    var weight = 0f
    var color: String? = null
    var gender: String? = null
    var tier: String? = null
    var requiredItem: String? = null

    fun matchingAbility(abilityId: String, or: String = abilityId): String {
        abilities.forEach { a -> if (a.toId() == abilityId) return a }
        if (hiddenAbility?.toId() == abilityId) return hiddenAbility!!
        return or
    }

}